### PR TITLE
feat: support indicatorsRender

### PR DESCRIPTION
**Pull request by li-jia-nan**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [x] New feature
- [x] Bug fix
- [x] Site / documentation update
- [x] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 💡 Background and solution

1. 修复了 `stepRender` 属性不工作的问题
2. `stepRender` api 重命名为 `indicatorsRender`
3. `ant-tour-slider` className 重命名为 `ant-tour-indicators`

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | feat: support indicatorsRender |
| 🇨🇳 Chinese | feat: 支持自定义指示器 |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by MadCcc** [comment]
来个简单 demo 吧～

File: components/tour/demo/indicator.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+## zh-CN
+
+自定义指示器。
+
+## en-US
+
+custom indicator.
```
Comment (MadCcc) [inline components/tour/demo/indicator.md:7]:
```suggestion
Custom indicator.
```

**Comment by MadCcc** [comment]
标题好长。。改成 feat 就好了

File: components/tour/index.en-US.md [hunk 2]
```diff
@@ -37,6 +38,7 @@ Use when you want to guide users through a product.
 | onChange | Callback when the step changes. Current is the previous step | `(current: number) => void` | - |  |
 | current | What is the current step | `number` | - |  |
 | scrollIntoViewOptions | support pass custom scrollIntoView options | `boolean \| ScrollIntoViewOptions` | `true` | 5.2.0 |
+| indicatorsRender | custom indicator | `(current: number, total: number) => ReactNode` | - |  |
```
Comment (MadCcc) [inline components/tour/index.en-US.md:41]:
```suggestion
| indicatorsRender | custom indicator | `(current: number, total: number) => ReactNode` | - | 5.2.0 |
```

File: components/tour/index.zh-CN.md [hunk 3]
```diff
@@ -39,6 +40,7 @@ demo:
 | onChange | 步骤改变时的回调，current 为当前前的步骤 | `(current: number) => void` | - |  |
 | current | 当前处于哪一步 | `number` | - |  |
 | scrollIntoViewOptions | 是否支持当前元素滚动到视窗内，也可传入配置指定滚动视窗的相关参数 | `boolean \| ScrollIntoViewOptions` | `true` | 5.2.0 |
+| indicatorsRender | 自定义指示器 | `(current: number, total: number) => ReactNode` | - |  |
```
Comment (MadCcc) [inline components/tour/index.zh-CN.md:43]:
```suggestion
| indicatorsRender | 自定义指示器 | `(current: number, total: number) => ReactNode` | - | 5.2.0 |
```

**Review by MadCcc — APPROVED** [review]

**Comment by yoyo837** [comment]
怎么看着像重命名API?

**Comment by li-jia-nan** [comment]
> 怎么看着像重命名API?

完整log是：修复了 stepRender 属性不工作的问题、stepRender api 重命名为 indicatorsRender、ant-tourant-tour-slider className 重命名为 ant-tour-indicatorsant-tour-indicators

**Comment by yoyo837** [comment]
看上去混入了其他的commit

**Comment by li-jia-nan** [comment]
> 建议写在“💡 Background and solution”的位置

updated~

**Comment by li-jia-nan** [comment]
> 看上去混入了其他的commit

改分支的时候把 master 的 diff 带过来了，我已经手动清理了~

**Comment by yoyo837** [comment]
image

你清理的是文件, commit还是在的

**Comment by yoyo837** [comment]
我试了一下, 好像也不对.

原始分支我备份了 `zhishi-backup`

**Comment by li-jia-nan** [comment]
- #40613

**Pull request closed without merging**